src/modules/px4iofirmware/safety_button.cpp

```diff
@@ -93,26 +86,23 @@ failsafe_led_init(void)
 }
 
 static void
-safety_check_button(void *arg)
+safety_button_check(void *arg)
 {
-	const bool safety_button_pressed = BUTTON_SAFETY;
+	const bool safety_button_pressed = px4_arch_gpioread(GPIO_BTN_SAFETY);
 
 	/* Keep safety button pressed for one second to turn off safety
 	 *
 	 * Note that safety cannot be turned on again by button because a button
 	 * hardware problem could accidentally disable it in flight.
```

Comment:
Can you update this?
- it's just generating an event now
- the logic to prevent turning on safety again is in commander now

Reply:
Thx, updated.